ESCPOS_NET/Emitters/BaseCommandEmitter/ImageCommands.cs

```diff
@@ -58,100 +55,35 @@ public byte[] BufferImage(byte[] image, int maxWidth = -1, bool isLegacy = false
                     break;
             }
 
-            using (Image<Rgba32> img = Image.Load(image))
+            int width;
+            int height;
+            byte[] imageData;
+            using (var img = Image.Load(image))
             {
-                int width = img.Width;
-                int height = img.Height;
-                // Use native width/height of image without resizing if maxWidth is default value.
-                if (maxWidth != -1)
-                {
-                    width = maxWidth;
-                    // Get closest height that's the same aspect ratio as the width.
-                    height = (int)(maxWidth * (Convert.ToDouble(img.Height) / img.Width));
-                    img.Mutate(x => x.Resize(width, height));
-                }
-                byte heightL = (byte)(height);
-                byte heightH = (byte)(height >> 8);
-
-                if (isLegacy)
-                {
-                    // TODO: if there's a remainder, need to add 1?
-                    var byteWidth = width / 8;
-                    if (width % 8 != 0)
-                    {
-                        byteWidth += 1;
-                    }
-                    byte widthL = (byte)byteWidth;
-                    byte widthH = (byte)(byteWidth >> 8);
-                    imageCommand.Append(new byte[] { Cmd.GS, Images.ImageCmdLegacy, 0x30, 0x00, widthL, widthH, heightL, heightH });
-                }
-                else
-                {
-                    byte widthL = (byte)(width);
-                    byte widthH = (byte)(width >> 8);
-                    imageCommand.Append(new byte[] { 0x30, 0x70, 0x30, 0x01, 0x01, colorByte, widthL, widthH, heightL, heightH });
-                }
-
-
-                // TODO: test making a List<byte> if it's faster than using ByteArrayBuilder.
-                double[] ditherErrorsNext = new double[width + 2];
-                // Bit pack every 8 horizontal bits into a single byte.
-                for (int y = 0; y < img.Height; y++)
-                {
-                    // Create an array for capturing dither errors
-                    double[] ditherErrorsCurrent = ditherErrorsNext;
-                    ditherErrorsNext = new double[width + 2];
+                imageData = img.ToSingleBitPixelByteArray(maxWidth: maxWidth == -1 ? (int?)null : maxWidth);
+                height = img.Height;
+                width = img.Width;
+            }
 
-                    Span<Rgba32> pixelRowSpan = img.GetPixelRowSpan(y);
-                    byte buffer = 0x00;
-                    int bufferCount = 7;
-                    for (int x = 0; x < img.Width; x++)
-                    {
-                        // Determine if pixel should be colored in.
-                        var pixelValue = ((0.30 * pixelRowSpan[x].R) + (0.59 * pixelRowSpan[x].G) + (0.11 * pixelRowSpan[x].B) + (ditherErrorsCurrent[x]));
```

Comment:
this algorithm is doing a weighted RGB conversion.  Is the black/white diffuser in the new implementation doing this compensation?  It looked like the threshold is set at 0.5 so this would need to be applied before grayscaling.

Reply:
I proposed [Stucki](https://github.com/SixLabors/ImageSharp/blob/72a820e0d1a11fc5fe397620b34910600512e842/src/ImageSharp/Processing/Processors/Dithering/DITHER.TXT#L318) filter [ErrorDispersion](https://github.com/SixLabors/ImageSharp/blob/72a820e0d1a11fc5fe397620b34910600512e842/src/ImageSharp/Processing/Processors/Dithering/DITHER.TXT#L248) algorithm because it is considered better in the ImageSharp [documentation](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Processing/Processors/Dithering/DITHER.TXT). 

I compared the printed result with the current algorithm and the new one provided by ImageSharp, there is almost no difference. Do you think another filter/transformation should be applied?

Reply:
I'll take a look with some test images as well, if they all look similar then I don't see any issue with this.